torch/_dynamo/output_graph.py

```diff
@@ -1413,6 +1415,21 @@ def compile_subgraph(
 
         from .decorators import disable
 
+        if has_user_objects():
```

Comment:
Add some comments on the index_to_user_object_weakref being a sync point between the pre-graph bytecode and compiled object.